DEPRECATIONS.md

```diff
@@ -0,0 +1,67 @@
+# Deprecations & Breaking Changes in KEDA
+
+This document describes how KEDA handles deprecations and breaking changes and should provide clarity to KEDA end-users on what to expect when upgrading to a new version of KEDA.
+
+If you are interested in the current open deprecations, we recommend checking our [GitHub Discussions](https://github.com/kedacore/keda/discussions/categories/deprecations) or the open breaking changes](https://github.com/kedacore/keda/issues?q=is%3Aopen+label%3Abreaking-change+sort%3Aupdated-desc).
+
+## Our versioning strategy
+
+KEDA follows the [Semantic Versioning](https://semver.org/) strategy and can only introduce breaking changes in new major versions.
+
+However, depending on the nature of the change, we may introduce deprecations in existing minor versions. These changes always must be backwards-compatible.
+
+Depending on the area of the change, we might not need to wait for the next major version of KEDA to introduce a breaking change.
+
+### KEDA runtime
+
+The KEDA runtime, which consists of the operator and metric server, is not allowed to make breaking changes unless the major version is incremented.
+
+There are no exceptions to this rule.
+
+### Custom Resource Definitions (CRDs)
+
+Our custom resource definitions (CRDs) have a separate versioning scheme than the KEDA runtime given they have an `apiVersion`. For this, we follow the [official Kubernetes API versioning policy](https://kubernetes.io/docs/reference/using-api/#api-versioning).
+
+KEDA is allowed to make breaking changes to the CRDs, when a new `apiVersion` is introduced.
+
+However, the KEDA runtime must support all current API versions until the next major version of KEDA.
+
+### Autoscaling triggers and their metadata
+
+Until KEDA introduces a [versioning scheme for autoscaling triggers](https://github.com/kedacore/keda/issues/613), it is not allowed to make breaking changes for triggers.
```

Comment:
If we finally change the deprecation policy, this line should be modified because we will allow breaking changes with conditions

Reply:
> If we finally change the deprecation policy

What are you referring to here? This policy or another one that I'm not are of?

Reply:
I'm referring to this https://github.com/kedacore/governance/pull/70/files#r984894940.
`it is not allowed to make breaking changes for triggers.` sounds too "categorical" because if we support deprecated fields for 2 versions, we will introduce breaking changes. 
In the same way, even we introduce scaler versions, if we are going to maintain each scaler version for 2 versions, we will introduce breaking changes.

Reply:
Even if we don't remove old scaler version after 2 KEDA versions, we have talked about not maintain a bunch of old versions for each trigger, so eventually we are going to introduce a breaking change related with this because we don't have plans for KEDA v3

Reply:
Updated after our discussion